src/app/components/Layout/TopBarLayout.tsx

```diff
@@ -33,6 +34,7 @@ export const TopBarLayout = ({ pageComponent, isTopBarVisible = true }: TopBarLa
         {pageComponent}
         <Login />
         <Signup />
+        <ForgotPasswordModal />
```

Comment:
I think there is no need to put this modal at layout level because it should be accesible only from Login, What do you think about it?

Reply:
In that case we can remove its open handler from context and implement only a local state for it

Reply:
Also, please check there are some conflicts in this file

Reply:
Yeah, no problem!
Regarding the conflicts it's weird because GitHub doesn't notify me, I'll merge with recent changes locally to check it.